### ISSUE TITLE
Automatic Background HTTP Connection Requests.

### ISSUE THREAD

**Issue by almonawar** [nvaccess/nvda#16875]
### Is your feature request related to a problem? Please describe.

I am envisioning the following behavior to be a problem, even though it is not, and NVDA is behaving as designed.

 NVDA connects to the internet to retrieve/check the hash of add-ons list, either available or updatable, every time it starts, reverts to saved configuration, or opens add-on store. This behavior occurs even when NVDA in Secure Mode and add-ons are disabled, which does not make sense.

### Describe the solution you'd like

 I envision an alternative behavior where NVDA will only check that hash value when the add-on store is opened and/or when the user initiate a process to check for add-ons update or availability. At least, in a worst case scenario, to give the user the ability to modify this behavior without resorting to block NVDA from accessing the internet as the user may need to use existing features or add-ons that requires connection to the internet.

### Describe alternatives you've considered

 I attempted to disable add-on store all together, but yet NVDA attempted to connect to the internet. Similarly, with the Secure Mode the same thing occurs. I am blocking internet access for now.

### Additional context

Regular user, and in a corporate environment. Utilizing Malwarebytes Windows Firewall Control to monitor connections, and Fiddler Classic to trace HTTP (s) network traffic.

**Comment by seanbudd** [comment]
@almonawar is there a reason why this was closed?

**Comment by almonawar** [comment]
No. Apparently NVDA needs to validate that value to function correctly and this feature request is invalid.

**Comment by seanbudd** [comment]
I don't believe this is true. I think this issue is valid
